program-test/tests/warp.rs

```diff
@@ -95,3 +106,145 @@ async fn custom_warp() {
         ProgramTestError::InvalidWarpSlot,
     );
 }
+
+#[tokio::test]
+async fn rent_collected_from_warp() {
+    let program_id = Pubkey::new_unique();
+    // Initialize and start the test network
+    let program_test = ProgramTest::default();
+
+    let mut context = program_test.start_with_context().await;
+    let account_size = 100;
+    let keypair = Keypair::new();
+    let account_lamports = Rent::default().minimum_balance(account_size) - 100; // not rent exempt
+    let instruction = system_instruction::create_account(
+        &context.payer.pubkey(),
+        &keypair.pubkey(),
+        account_lamports,
+        account_size as u64,
+        &program_id,
+    );
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &keypair],
+        context.last_blockhash,
+    );
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+    let account = context
+        .banks_client
+        .get_account(keypair.pubkey())
+        .await
+        .expect("account exists")
+        .unwrap();
+    assert_eq!(account.lamports, account_lamports);
+
+    // Warp forward and see that rent has been collected
+    // This test was a bit flaky with one warp, but two warps always works
+    context.warp_to_slot(DEFAULT_DEV_SLOTS_PER_EPOCH).unwrap();
```

Comment:
Just a nit that resolving in a follow-up sometime seems fine: relying on `DEFAULT_DEV_SLOTS_PER_EPOCH` here is yucky.   Instead how about adding `fn genesis_config(&self) -> &GenesisConfig` method to `ProgramTestContext` so the test can do this: `context.genesis_config().epoch_schedule.slots_per_epoch`